temporian/implementation/numpy/operators/boolean/base.py

```diff
@@ -0,0 +1,84 @@
+from typing import Dict, Union
+from abc import abstractmethod
+
+import numpy as np
+
+from temporian.core.operators.boolean.base_scalar import (
+    BaseBooleanScalarOperator,
+)
+from temporian.core.operators.boolean.base_feature import (
+    BaseBooleanFeatureOperator,
+)
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.feature import NumpyFeature
+from temporian.implementation.numpy.operators.base import OperatorImplementation
+
+
+class BaseBooleanNumpyImplementation(OperatorImplementation):
+    """Abstract base class to implement common logic of numpy implementation of
+    boolean operators."""
+
+    def __init__(
+        self,
+        operator: Union[BaseBooleanScalarOperator, BaseBooleanFeatureOperator],
+    ) -> None:
+        assert isinstance(
+            operator, (BaseBooleanScalarOperator, BaseBooleanFeatureOperator)
+        ), (
+            "Expected operator to be of type BaseBooleanScalarOperator or"
+            f" BaseBooleanFeatureOperator, got {type(operator)}"
+        )
+        super().__init__(operator)
+
+    def __call__(
+        self, event_1: NumpyEvent, event_2: NumpyEvent = None
+    ) -> Dict[str, NumpyEvent]:
+        # if operator is a Scalar Operator, value is the scalar
```

Comment:
I would separate the scalar and feature in two different implementation.

Note that the feature implementation could derive the arithmetic base class.
Instead, what about creating an arithmetic_scalar base class? This way, we can then easily implement operator such as "tp.add(a["x"], 5).